src/ln/channelmonitor.rs

```diff
@@ -1196,13 +1206,98 @@ impl ChannelMonitor {
 					txn_to_broadcast.push(spend_tx);
 				}
 			}
-		} else {
-			//TODO: For each input check if its in our remote_commitment_txn_on_chain map!
 		}
 
 		(txn_to_broadcast, (commitment_txid, watch_outputs))
 	}
 
+	/// Attempst to claim a remote HTLC-Success/HTLC-Timeout s outputs using the revocation key
+	fn check_spend_remote_htlc(&self, tx: &Transaction, commitment_number: u64) -> Vec<Transaction> {
```

Comment:
Was refactored mainly to avoid double-locking of remote_commitment_tx_on_chain as block_connected has only them to match HTLC transactions, so a part of old commit was moved there.